### ISSUE TITLE
Servers across the networks?

### ISSUE THREAD

**Issue by mesb1** [anacrolix/dms#153]
Hi.

Is it possible to have such functionality:

Let's say we have a network1 with a dms1 and few dlna servers that works only for this network as upnp declares.
Can we have some additional dms2 installation on network2 that connected via vpn for network1 but just retransmitting all the data from network1 related to dms1 and all other dlna servers from that network?

**Comment by anacrolix** [comment]
Yes, I think it would require some fancy network interface configuration. 

In particular for SSDP you would need to forward and possible translate fields in the messages similar to how a NAT works. You might also have to NAT the following queries from network2. So essentially an SSDP NAT I guess, the mappings it generates would allow it to act as a proxy into network1.

The easier answer is obviously to have everyone in network1 and 2 be in a VPN together.

**Comment by fenio** [comment]
@mesb1 It's more a job for proxy rather than for DMS itself.
You can use https://github.com/Nic0w/dlnaproxy for that or my fork of it which fixes bunch of issues: https://github.com/fenio/dlnaproxy

**Comment by anacrolix** [comment]
I'm going to do the unthinkable: there's nothing else to solve here and nothing to be done. @fenio has delivered a great solution.

**Comment by mesb1** [comment]
> [@mesb1](https://github.com/mesb1) It's more a job for proxy rather than for DMS itself. You can use https://github.com/Nic0w/dlnaproxy for that or my fork of it which fixes bunch of issues: https://github.com/fenio/dlnaproxy

Sounds cool. 
But for your release or the original one there is no dlnaproxy binary.
I was hope for some mips, mipsel, armv7l and aarch64 binaries to run it with router via entware/optware.
There is not possible to run docker on the router itself.

Also right now it was using not only minidlna, but xupnpd also.

**Comment by fenio** [comment]
@mesb1 LOL… this issue is over a year old and it looks like you got stuck waiting for some kind of solution 😉
I showed up here pretty much by pure chance.

Given how niche the architectures you mentioned are, I didn’t even consider them for dlnaproxy, since I assumed users would be running it on something less exotic.

But specially for you I built versions for amd64, arm64, mips, mipsel, and arm7l. If you need any others, as long as they’re listed at https://rust-lang.github.io/rustup-components-history it shouldn’t be a problem.

I didn’t make a dlnaproxy release because, as I mentioned before, this is a fork and not my project, but you can find the binaries at https://github.com/fenio/dlnaproxy/actions/runs/20758323475

**Comment by mesb1** [comment]
> fork and not my project,

Sinse original one doesn't updated anymore, you can atleast open issues section to receive some feedback)

**Comment by mesb1** [comment]
> but you can find the binaries at https://github.com/fenio/dlnaproxy/actions/runs/20758323475

Oh man, thank you very much.

**Comment by mesb1** [comment]
> it shouldn’t be a problem

It shouldn't, atleast for original x86-64.
While crosscompile was a pain for last couple of hours extensive googling(

**Comment by fenio** [comment]
@mesb1 Ok since I wanted to get rid of constant bothering about how ahead I'm compared to forked repo I disconnected it and recreated from scratch so forget about old links. Everything is now here: https://github.com/fenio/dlna-proxy
And there's a normal release with binaries and ability to create issues.

**Comment by anacrolix** [comment]
Happy to link to your proxy if you think it's worthwhile. Make a PR to add it to the README or something.

**Comment by fenio** [comment]
@anacrolix Thanks for the offer. I created #179 for that.
